client/state/posts/actions.js

```diff
@@ -68,3 +71,36 @@ export function requestSitePosts( siteId, query = {} ) {
 		} );
 	};
 }
+
+/**
+ * Triggers a network request to fetch a specific post from a site.
+ *
+ * @param  {Number}   siteId Site ID
+ * @param  {Number}   postId Post ID
+ * @return {Function}        Action thunk
+ */
+export function requestSitePost( siteId, postId ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: POST_REQUEST,
+			siteId,
+			postId
+		} );
+
+		return wpcom.site( siteId ).post( postId ).get().then( ( post ) => {
+			dispatch( receivePost( post ) );
+			dispatch( {
+				type: POST_REQUEST_SUCCESS,
+				siteId,
+				postId
+			} );
+		} ).catch( ( error ) => {
```

Comment:
There's a real subtle diff between the two cases. By using catch after the then, you're also catching any errors that the first then handler throws. If you use the `then( yay, boo )` construct, `boo` will never see errors that `yay` throws. So it really depends on what you're trying to capture and who's going to consume the Promise.

see http://www.html5rocks.com/en/tutorials/es6/promises/#toc-error-handling